lib/mvi/responses/find_candidate.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+require_relative 'base'
+
+module MVI
+  module Responses
+    class FindCandidate < Base
+      mvi_endpoint :prpa_in201306_uv02
+
+      def initialize(response)
+        super(response)
+        @subject = @original_body.dig(:control_act_process, :subject)
+      end
+
+      def body
+        return nil unless @subject
+        patient = @subject.dig(:registration_event, :subject1, :patient)
+        name = parse_name(patient.dig(:patient_person, :name))
+        {
+          status: patient.dig(:status_code, :@code),
+          given_names: name[:given],
+          family_name: name[:family],
+          gender: patient.dig(:patient_person, :administrative_gender_code, :@code),
+          birth_date: patient.dig(:patient_person, :birth_time, :@value),
+          ssn: parse_ssn(patient.dig(:patient_person, :as_other_i_ds))
+        }.merge(map_correlation_ids(patient[:id]))
+      end
+
+      private
+
+      # name can be a hash or an array of hashes with extra unneeded details
+      # given may be an array if it includes middle name
+      def parse_name(name)
+        name = [name] if name.is_a? Hash
+        name_hash = [*name].first
+        given = [*name_hash[:given]].map(&:capitalize)
+        family = name_hash[:family].capitalize
+        { given: given, family: family }
+      rescue => e
+        Rails.logger.warn "MVI::Response.parse_name failed: #{e.message}"
+        { given: nil, family: nil }
+      end
+
+      # other_ids can be hash or array of hashes
+      def parse_ssn(other_ids)
+        other_ids = [other_ids] if other_ids.is_a? Hash
+        ssn_id = other_ids.select { |id| id.dig(:id, :@root) == '2.16.840.1.113883.4.1' }
+        return nil if ssn_id.empty?
+        ssn_id.first.dig(:id, :@extension)
+      rescue => e
+        Rails.logger.warn "MVI::Response.parse_ssn failed: #{e.message}"
+        nil
+      end
+
+      # MVI correlation id source id relationships:
+      # {source id}^{id type}^{assigning authority}^{assigning facility}^{id status}
+      # NI = national identifier, PI = patient identifier
+      def map_correlation_ids(ids)
```

Comment:
I don't like large overbearing comments - but it feels warranted here.  This method may as well be written in hieroglyphics - I have no clue what's going on.